### PR TITLE
Fix syntax of second concurrency example

### PR DESCRIPTION
**Pull request by AnyTimeTraveler**
Since I could not understand what the Java code was even meant to do, I took the Kotlin code and rewrote it in Java as I would have written it if I had never seen the Java side of it.

### CODE REVIEW

File: code/java/concurrency/02.java [hunk 1]
```diff
@@ -1,19 +1,13 @@
...
-                                if (ty2.isFailure && ty1.isSuccess)
-                                    p.failed((ty2 as Failure).exception)
-                                else
-                                    p.complete(ty1)
-                            }
-                    }
-                    p
-            }
-        }
+public <A> CompletableFuture<A> inTransaction(DatabaseFunction<A> fun) {
```
Comment (driver733) [inline code/java/concurrency/02.java]: "A" should be replaced with "T", as it's conventionally used to indicate type.

Comment (driver733) [inline code/java/concurrency/02.java]: The code should be formatter properly (Indent nested blocks with 4 spaces, move method and lambda params to the next line).

Comment (AnyTimeTraveler) [reply code/java/concurrency/02.java]: I agree on the indentation for the inside of the lambda functions. Will fix that.
What do you mean with moving stuff to the next line?
I've oriented myself on the syntax of the [CollectingAndThen](https://www.kotlinvsjava.com/collections.html) example on the page and don't see anything that needs to be changed there.

Please be more specific.

Comment (AnyTimeTraveler) [reply code/java/concurrency/02.java]: I can change that, but the Kotlin code also uses A as it's type parameter.
Should both be changed, then?

**Comment by driver733** [comment]
@AnyTimeTraveler Merge conflicts must be resolved before merging.

**Pull request merged**